recipes/glib/all/conanfile.py

```diff
@@ -183,47 +203,51 @@ def _patch_sources(self):
             "'share'",
             "'res'",
         )
-        if self.settings.os != "Linux":
-            tools.replace_in_file(
-                os.path.join(self._source_subfolder, "meson.build"),
-                "if cc.has_function('ngettext'",
-                "if false #cc.has_function('ngettext'",
-            )
+
+    def _patch_pkgconfig(self):
+        with chdir(self, self.generators_folder):
+            for filename in glob.glob("*.pc"):
+                self.output.info("processing %s" % filename)
+                content = load(self, filename)
+                results = re.findall(r"-F (.*)[ \n]", content)
+                for result in results:
+                    if not os.path.isdir(result):
+                        self.output.info("removing bad framework path %s" % result)
+                        content = content.replace("-F %s" % result, "")
+                save(self, filename, content)
 
     def build(self):
         self._patch_sources()
-        with tools.environment_append(
-            VisualStudioBuildEnvironment(self).vars
-        ) if is_msvc(self) else tools.no_op():
-            meson = self._configure_meson()
-            meson.build()
+        self._patch_pkgconfig()
+        meson = Meson(self)
+        meson.configure()
+        meson.build()
 
     def _fix_library_names(self):
         if self.settings.compiler == "Visual Studio":
-            with tools.chdir(os.path.join(self.package_folder, "lib")):
+            with chdir(self, os.path.join(self.package_folder, "lib")):
                 for filename_old in glob.glob("*.a"):
                     filename_new = filename_old[3:-2] + ".lib"
                     self.output.info(f"rename {filename_old} into {filename_new}")
                     shutil.move(filename_old, filename_new)
 
     def package(self):
-        if scm.Version(self.version) < "2.73.0":
-            self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        if Version(self.version) < "2.73.0":
+            self.copy(pattern="COPYING", dst="licenses", src=self.source_folder)
```

Comment:
```suggestion
            copy(self, pattern="COPYING", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
```
from conan.tools.file import copy